zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServerMain.java

```diff
@@ -192,9 +192,7 @@ public void runFromConfig(ServerConfig config) throws IOException, AdminServerEx
             if (secureCnxnFactory != null) {
                 secureCnxnFactory.join();
             }
-            if (zkServer.canShutdown()) {
-                zkServer.shutdown(true);
-            }
+            zkServer.shutdown(true);
```

Comment:
We shouldn't check this in more places than one, and that should be inside the server itself.